src/function/execute.rs

```diff
@@ -38,53 +38,122 @@ where
             },
         });
 
-        // If we already executed this query once, then use the tracked-struct ids from the
-        // previous execution as the starting point for the new one.
-        if let Some(old_memo) = &opt_old_memo {
-            active_query.seed_tracked_struct_ids(&old_memo.revisions.tracked_struct_ids);
-        }
+        let mut iteration_count: u32 = 0;
 
-        // Query was not previously executed, or value is potentially
-        // stale, or value is absent. Let's execute!
-        let database_key_index = active_query.database_key_index;
-        let id = database_key_index.key_index;
-        let value = match Cycle::catch(|| C::execute(db, C::id_to_input(db, id))) {
-            Ok(v) => v,
-            Err(cycle) => {
+        // Our provisional value from the previous iteration, when doing fixpoint iteration.
+        // Initially it's set to None, because the initial provisional value is created lazily,
+        // only when a cycle is actually encountered.
+        let mut opt_last_provisional: Option<&Memo<<C as Configuration>::Output<'db>>> = None;
+
+        loop {
```

Comment:
Can we branch on the cycle strategy here? There is a lot of work that I don't think needs to happen for `Panic` strategies that non cycle supporting queries currently pay (and given the branching would happen on a `const` it should optimized out)